ReactiveFeedback/Store.swift

```diff
@@ -0,0 +1,62 @@
+import ReactiveSwift
+
+open class Store<State, Event> {
+    public let state: Property<Context<State, Event>>
+
+    private let input = Feedback<State, Event>.input()
+    private let send: (Event) -> Void
+
+    public init<S: DateScheduler>(
+        initial: State,
+        reducer: @escaping Reducer<State, Event>,
+        feedbacks: [Feedback<State, Event>],
+        scheduler: S
+    ) {
+        self.send = { _ in }
+        self.state = Property(
+            initial: Context(state: initial, send: self.input.observer),
+            then: SignalProducer.system(
+                initial: initial,
+                scheduler: scheduler,
+                reduce: reducer,
+                feedbacks: feedbacks.appending(self.input.feedback)
+            )
+            .map { [input] state in
+                Context(state: state, send: input.observer)
+            }
+            .skip(first: 1)
+        )
+    }
+
+    private init(state: Property<Context<State, Event>>, send: @escaping (Event) -> Void) {
+        self.state = state
+        self.send = send
+    }
+
+    open func send(event: Event) {
+        input.observer(event)
+        send(event)
+    }
```

Comment:
In this class's instance variable, there are:

- `let state: Property<Context<State, Event>>` that has `state.value.send: (Event) -> Void` 
- `let send: (Event) -> Void`

which are duplicate and are both observers are called in this `func send(event: Event)`.

Can you explain why this duplicated structure is needed?

Reply:
`send` is used only via `func view` when we view into sub system

Reply:
@sergdort 
I think I was a bit mixed up with:

1. Duplicated `(Event) -> Void` member variables (https://github.com/babylonhealth/ReactiveFeedback/pull/49#discussion_r356987191)
2. Multiple observer/send calls (https://github.com/babylonhealth/ReactiveFeedback/pull/49#discussion_r357191319)

For 2., as you mentioned, I now understand as:

```swift
input.observer(event)

// Forward event delivery to parent store, where @pointfreeco is doing inside reducer call.
send(event)
```

So, `let send` could probably be renamed as e.g. `forward` to disambiguate from `func send(event:)` [nit].

But for 1., I still don't get **why `state.value.send` needs to be stored inside `Property` as a reactive stream representation**.
I know that `state.value.send` will be called inside the local renderer, but **this method will not be changed** during the state changes and re-render, so IMO it's not worth storing to be delivered in reactive stream.
(It is like making `Observable<Observer<Event>>` which sends fixed `state.value.send` observers over time, which I have never seen this kind of signature before)

So, I think there could be some improvements around 1. to remove duplication to make overall code a bit more simpler.
But I'm also fine with current approach as I think it won't probably cause any major issues.
(Writing some tests will definite solve this uncertainty.)